src/string_builder.hpp

```diff
@@ -6,29 +6,40 @@
 #include <format>
 
 namespace experimental_json_builder {
+template <typename T> concept arithmetic = std::is_arithmetic_v<T>;
 
-class StringBuilder {
+class StringBuilder final {
 public:
-  StringBuilder(size_t initial_capacity = 2629189250)
+  // By default, we allocate 1 MB.
+  StringBuilder(size_t initial_capacity = 1048576)
       : buffer(new char[initial_capacity]), position(0),
         capacity(initial_capacity) {}
-  inline void append(double v) {
-    auto [ptr, ec] =
+
+  template <arithmetic number_type>
+  inline void append(number_type v) {
+    auto result =
+        std::to_chars(buffer.get() + position, buffer.get() + capacity, v);
+    if(result.ec != std::errc()) {
+      constexpr size_t max_number_size = 20;
+      capacity_check(max_number_size);
+      result =
         std::to_chars(buffer.get() + position, buffer.get() + capacity, v);
-    (void)ec; // no error handling
-    position = ptr - buffer.get();
+    }
+    position = result.ptr - buffer.get();
   }
 
-  inline void append(char c) { buffer[position++] = c; }
+  inline void append(char c) { capacity_check(1); buffer[position++] = c; }
 
   void reset() { position = 0; }
 
   inline void append_escaped_json(std::string_view input) {
+    capacity_check(6*input.size());
```

Comment:
I get the +2 between escaped and quoted_escaped, but I am not sure I understand the multiplication by 6. Can you explain the rationale here?

Reply:
@FranciscoThiesen In the worst case a character can become 6 characters... right? So the byte value 4 might become `\u0004`? The idea is to avoid having to do two passes and instead just compute very quickly the worst possible bound.

Reply:
Makes sense! Thanks for clarifying